### ISSUE TITLE
Invalid type declaration after upgrading to TS 4.7

### ISSUE THREAD

**Issue by DebkanchanSamadder** [firebase/firebase-admin-node#1726]
### [READ] Step 1: Are you in the right place?

  * For issues related to __the code in this repository__ file a Github issue.
  * If the issue pertains to Cloud Firestore, read the instructions in the "Firestore issue"
    template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

  * Operating System version: MacOS Monterey 12.3.1
  * Firebase SDK version: 10.2.0
  * Firebase Product: app, firestore
  * Node.js version: 16
  * NPM version: 8.5.5

#### Steps to reproduce:

After upgrading to Typescript 4.7, `tsc` throws the following error for all firebase admin SDK imports

```bash
error TS7016: Could not find a declaration file for module 'firebase-admin/firestore'. '/Users/debkanchan/Library/Mobile Documents/com~apple~CloudDocs/Work/Ride/Code/Backend/services/wallet-service/node_modules/firebase-admin/lib/esm/firestore/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/firebase-admin` if it exists or add a new declaration (.d.ts) file containing `declare module 'firebase-admin/firestore';`

2 import { Firestore, FieldValue, getFirestore } from "firebase-admin/firestore";
```

#### Relevant Code:
tsconfig.json

```jsonc
{
	"extends": "@tsconfig/node16/tsconfig.json",
	"compilerOptions": {
		"module": "Node16",
		"outDir": "build",
		"removeComments": true
		// "noImplicitAny": false
	},
	"compileOnSave": true,
	"include": ["src"]
}
```

package.json
```jsonc
{
	"type": "module",
	"engines": {
		"node": "16"
	},
       ...
	"dependencies": {
		"firebase-admin": "^10.2.0",
                ...
	},
	"devDependencies": {
		"@tsconfig/node16": "^1.0.2",
                 ...
		"ts-node": "^10.8.0",
		"typescript": "^4.7.2"
	}
}

```

**Comment by lahirumaramba** [comment]
Hi @DebkanchanSamadder Thank you reporting this issue! I think we need to point to the correct typing definitions in the `package.json` using the `types` field. https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#package-json-exports-imports-and-self-referencing

I will create a PR for this. Thanks!

**Comment by DebkanchanSamadder** [comment]
Hey @lahirumaramba , Any updates on this? This is a blocker for moving to New ts version

**Comment by lotus-x** [comment]
Yeah same here, it would be great to have an updated version with the fix.

**Comment by lahirumaramba** [comment]
This should be fixed in v10.3.0. Thanks!
